buildpacks/go/bin/build

```diff
@@ -23,9 +23,6 @@ EOF
 
 cp $build_dir/bin/faas "$app_layer/faas"
 
-# REMOVE SOURCE CODE FROM FINAL CONTAINER
-rm -rf $build_dir/*
```

Comment:
@matejvasek I actually think this is a good thing to do. It keeps the source out of the runtime image. And more importantly, it removes func.yaml (which may contain sensitive data) from the run image.

Reply:
@lance there was an issue with deleting it while not being a root: stuff in `/workspace/go/pkg/mod/` seems to have `-r--r--r--` rights

Reply:
also do we remove those files in other BPs?